### ISSUE TITLE
[3.3.5][6.x][Ulduar] Core/Creature: XT-002 Deconstructor

### ISSUE THREAD

**Issue by Gargarensis** [TrinityCore/TrinityCore#16426]
When he expose his heart, it won't take double damage.
[Heart of the Deconstructor](http://www.wowhead.com/npc=33329/heart-of-the-deconstructor "Heart of the Deconstructor") should have the buff [Exposed Heart](http://www.wowhead.com/spell=63849/exposed-heart "Exposed Heart").

Rev: b1d0855624d1c83850fc36f5242437d7f7c1336b

**Comment by Keader** [comment]
i dont know why, but somtimes cast fails (maybe gcd for some spell)
Is a hack, but you can change CastSpell for AddAura, like this:

```diff
diff --git a/src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp b/src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp
index 8cb20ea..a041a25 100644
--- a/src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp
+++ b/src/server/scripts/Northrend/Ulduar/Ulduar/boss_xt002.cpp
@@ -374,7 +374,7 @@ class boss_xt002 : public CreatureScript
                     heart->CastSpell(heart, SPELL_HEART_OVERLOAD, false);
                     heart->CastSpell(me, SPELL_HEART_LIGHTNING_TETHER, false);
                     heart->CastSpell(heart, SPELL_HEART_HEAL_TO_FULL, true);
-                    heart->CastSpell(heart, SPELL_EXPOSED_HEART, false);    // Channeled
+                    heart->AddAura(SPELL_EXPOSED_HEART, heart);
                     heart->ChangeSeat(HEART_VEHICLE_SEAT_EXPOSED, true);
                     heart->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
                     heart->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_UNK_29);
```

But you lose Channel cast.

**Comment by Nyeriah** [comment]
It's already working fine. What happens is that the spell the heart channels can be interrupted.

**Comment by Keader** [comment]
can be closed @Nyeriah ? (i dont tested on current rev)

**Comment by Gargarensis** [comment]
On rev b1d0855624d1c83850fc36f5242437d7f7c1336b it does not even cast the spell.
I didn't test it on current rev. 
If you have tested it and it works, you can close the issue.

**Comment by Keader** [comment]
Maybe fixed by 3f886025a2559c4973ac260fc633ff74c2096932

**Comment by Rushor** [comment]
test on https://github.com/TrinityCore/TrinityCore/commit/3f886025a2559c4973ac260fc633ff74c2096932

**Comment by Keader** [comment]
Tested again in rev cb8f64cb63c9262717c7a019a581bd284579583c and still dont work

**Comment by Ercules76** [comment]
I've tried the  spell in npc_xt002_heart script, inside the DamageTaken() function, the spell work fine, at every hit the npc have the channeled buff. There's something inside the boss script which makes the annoyng thing.

**Comment by Yehonal** [comment]
The problem here, IMHO, is that the `heart->ChangeSeat(HEART_VEHICLE_SEAT_EXPOSED, true); ` has an async effect that starts the exposing animation. During this animation a channeled spell could be cancelled.

Some possible solution would be:

1) set a scheduled event  of few seconds , just after the ChangeSeat,  and cast the EXPOSE channeled spell then.

2) implement a function for vehicle scripts that is called just after the vehicle is "ready"

**Comment by ariel-** [comment]
Yehonal: see
```c++
virtual void PassengerBoarded(Unit* /*passenger*/, int8 /*seatId*/, bool /*apply*/) { }
```

maybe it's just the hook that you want

**Comment by Ercules76** [comment]
I resolve the fix with this, and work. 

`if (apply && who->GetEntry() == NPC_XT002_HEART && seatId == HEART_VEHICLE_SEAT_EXPOSED)
                    who->CastSpell(who, SPELL_EXPOSED_HEART, false);    //Channeled`

**Comment by Keader** [comment]
Test with: #17900

**Comment by Treeston** [comment]
Closed by #17900 in 1f83412.
